### PR TITLE
FUSETOOLS-2757 - added templates for 3 types of Syndesis/Ignite extensions

### PR DESCRIPTION
**Pull request by lhein**
# Pull Request Checklist

After this checklist is all checked or PR provides explanations for possible pass-through, please put the label "Ready for review" on the PR.

## General

- [x] Did you use the Jira Issue number in the commit comments?
- [x] Did you set meaningful commit comments on each commit?
- [x] Did you sign off all commits for this PR? (git commit -s -m "jira issue number - your commit comment")

## Functional

- [ ] Did the CI job report a successful build?
- [ ] Is the non-happy path working, too?

## Maintainability

- [ ] Are all Sonar reported issues fixed or can they be ignored?
- [x] Is there no duplicated code?
- [x] Are method-/class-/variable-names meaningful?

## Tests

- [ ] Are there unit-tests?
- [ ] Are there integration tests (or at least a jira to tackle these)?
- [ ] Do we need a new UI test?

## Legal

- [x] Have you used the correct file header copyright comment?
- [x] Have you used the correct year in the headers of new files?

### CODE REVIEW

**Comment by lhein** [comment]
based on #1288 and #1287

**Comment by lhein** [comment]
base for #1293

File: syndesis/plugins/org.fusesource.ide.syndesis.extension.ui/src/org/fusesource/ide/syndesis/extensions/ui/wizards/validation/SyndesisExtensionNameValidator.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ */
+public class SyndesisExtensionNameValidator implements IValidator {
+
+	/* (non-Javadoc)
+	 * @see org.eclipse.core.databinding.validation.IValidator#validate(java.lang.Object)
+	 */
+	@Override
+	public IStatus validate(Object value) {
+		String name = (String) value;
+		// validate the extension id
```
Comment (apupier) [inline syndesis/plugins/org.fusesource.ide.syndesis.extension.ui/src/org/fusesource/ide/syndesis/extensions/ui/wizards/validation/SyndesisExtensionNameValidator.java]: what about renaming "name" to "extensionId" and removing this comment?

File: syndesis/plugins/org.fusesource.ide.syndesis.extension.ui/src/org/fusesource/ide/syndesis/extensions/ui/wizards/validation/SyndesisExtensionIdValidator.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ */
+public class SyndesisExtensionIdValidator implements IValidator {
+
+	/* (non-Javadoc)
+	 * @see org.eclipse.core.databinding.validation.IValidator#validate(java.lang.Object)
+	 */
+	@Override
+	public IStatus validate(Object value) {
+		String id = (String) value;
+		// validate the extension id
```
Comment (apupier) [inline syndesis/plugins/org.fusesource.ide.syndesis.extension.ui/src/org/fusesource/ide/syndesis/extensions/ui/wizards/validation/SyndesisExtensionIdValidator.java]: what about renaming the "id" to "extensionId" and removing this comment?

**Comment by apupier** [comment]
testPR

File: syndesis/plugins/org.fusesource.ide.syndesis.extension.ui/src/org/fusesource/ide/syndesis/extensions/ui/wizards/validation/SyndesisExtensionNameValidator.java (hunk 1 above)
Comment (lhein) [reply syndesis/plugins/org.fusesource.ide.syndesis.extension.ui/src/org/fusesource/ide/syndesis/extensions/ui/wizards/validation/SyndesisExtensionNameValidator.java]: I think that just the comment is wrong.

**Pull request closed without merging**